pkg/pillar/cmd/zedagent/zedagent.go

```diff
@@ -876,6 +887,11 @@ func Run() {
 			subAppFlowMonitor.ProcessChange(change)
 			agentlog.CheckMaxTime(agentName, start)
 
+		case change := <-subVaultStatus.C:
```

Comment:
I think you should add this to the two previous select loops since we start reporting deviceinfo earlier in zedagent.